### ISSUE TITLE
[BUG] Postings - Check swipe has squared corners in the first row in a section

### ISSUE THREAD

**Issue by BDUCLM** [kirbydesign/designsystem#1895]
- [ ] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe how to reproduce the bug
1) Open the postings/transaction list
2) Swipe on the first row in a section
3) The check button is not with round corners - it is also squared on the right side of the list

## Which Kirby version was used?
4.0.8

## What was the expected behavior?
Corners are round.

## Add any screenshots
![IMG_0097](https://user-images.githubusercontent.com/94539419/142209355-74618a5e-6e4e-4731-b184-0c020a6db8d8.jpg)

## Please complete the following information:
- OS: iOS (iPad Air iOS 15.1)

## Are there any additional context?

<hr />

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/master/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Verification
To make sure the bug is not intended behaviour; it should be verified by a member of team Kirby before moving on to implementation. 

- [ ] Make sure the `NOT verified` label has been removed by a member of [team Kirby](https://github.com/kirbydesign/designsystem/blob/master/.github/SUPPORT.md#team-kirby); do not proceed until this is done

### Implementation 
The contributor who wants to implement this issue should: 

- [ ] Make sure you have read: "[Before you get coding](https://github.com/kirbydesign/designsystem/blob/master/.github/CONTRIBUTING.md/#before-you-get-coding)".
- [ ] Signal to others you are working on the issue by assigning yourself.
- [ ] Create a branch from the [master branch](https://github.com/kirbydesign/designsystem/tree/master) following our [branch naming convention](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Branch). 
- [ ] Create a test that reproduces the bug following guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)". 
- [ ] Write code to fix the bug

### Review
Once the issue has been implemented and is ready for review:

- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review). 
- [ ] Create a pull-request.

**Comment by RasmusKjeldgaard** [comment]
This is currently not happening on the cookbook, even though I have tried with a modified example with sections, that should mirror the above. 

![image](https://user-images.githubusercontent.com/42470636/142231254-83ec1ed5-7215-4dde-8f9f-47095e535920.png)

```
<h2>List with swipe actions:</h2>

<kirby-list [items]="items" showDivider="true" [swipeActions]="swipeActions" [getSectionName]="getSectionName">
  
  <kirby-list-section-header
    *kirbyListSectionHeader="let section"
    [title]="section"
  ></kirby-list-section-header>
  <div *kirbyListItemTemplate="let item" class="hellooo">
  <kirby-item>
    <div slot="outside">
      <div class="flag success" *ngIf="item.flagged"></div>
      <div class="flag warning" *ngIf="item.archived"></div>
    </div>
    <h3>{{ item.title }}</h3>
    <data slot="end" class="kirby-text-bold">{{ item.amount }}</data>
  </kirby-item>
</div>
</kirby-list>
```

But I can confirm it by checking the production app on my phone, and the problem does seem to be on our end. 

For some reason, the correct boundary class (`first`) is not added to the first `kirby-list-item` in a section, even though the corresponding 'last' class is correctly applied to the last item. If this class is manually added to the list item, it works as expected.

**Comment by mark-drastrup** [comment]
@RasmusKjeldgaard is this still an issue? :) I haven't been able to reproduce the issue.

**Comment by AgreSanGit** [comment]
We have figured out exactly why the problem occurs and why 'first' is not applied as a class on the first element in each section. If you search and access the file "kirbydesign-designsystem.mjs" in relationsbanken (no virtual scrolling) and go to code line 7082 you're able to see the method "_getBoundaryClass()" (see picture below). When index === 0 headTemplate is supposed to be undefined or null for the method to return 'first'. 

![Image](https://user-images.githubusercontent.com/112690169/202407139-e17e9817-13ea-46b3-8a4a-03156559747a.png)


But for some reason when index === 0  headTemplate is actually defined (see picture below). We would like to talk to a developer in relationsbanken and see the implementation to know why the headtemplate is being defined as it is and how the component is being used 😄 

![Image](https://user-images.githubusercontent.com/112690169/202407152-baf12563-94b6-4f48-b6c6-96aadfd798f3.png)

**Comment by KirstenHoffmeyer** [comment]
Hi,
We have solved the problem so this bug can be closed :-)

**Comment by AgreSanGit** [comment]
That's good to know! but we would still like to see the old implementation of the code where the issue occurs. So that we can recreate the problem on our end and gain some insight on how to guard against this issue for other developers and for future use.

**Comment by KirstenHoffmeyer** [comment]
Ok. You could invite jørgen vrou and Mikael Kristoffersen for a workthrough

**Comment by AgreSanGit** [comment]
We were not able to reproduce this bug on our end. But we were able to use fiddler to solve the bug in relationsbanken, by intercepting the response from a specific URL that fetches the file containing _getBoundaryClass. When intercepting we're replacing the javascript file before it compiles, with our own response with a modified version of that file which contains a new implementation of the _getBoundaryClass. By doing this we were able to observe that the new changes we made in the list component were able to fix this bug by not taking headerTemplate or footerTemplate into consideration when applying classes. We are considering making these changes to list component for a full bug fix.

**New implementaion of _getBoundaryClass:** 
![Screenshot 2022-11-25 at 13 23 57](https://user-images.githubusercontent.com/112690169/203985464-db16ab30-f53b-4f32-84a3-28eb8a03e7b9.png)

**End result:** 
![Screenshot 2022-11-25 at 13 27 18](https://user-images.githubusercontent.com/112690169/203986057-116452e5-ae96-4c14-a557-274d01eefdcc.png)
